### ISSUE TITLE
GetStreamURL() Returns URLs That Are Forbidden (403)

### ISSUE THREAD

**Issue by thenhawke** [kkdai/youtube#233]
The URLs returned from GetStreamURL() have started throwing an HTTP 403 when accessed. Code I'm using is below. This has been working up until very recently, perhaps YouTube has changed the way the URL must be crafted slightly? 

```
package main

import (
	"fmt"
	"log"

	"github.com/kkdai/youtube/v2"
)

func main() {
	client := youtube.Client{}

	//A few videos that 403
	//"djV11Xbc914"
	//"ZrOKjDZOtkA"
	//"ENXvZ9YRjbo"
	//"eFjjO_lhf9c"
	//"gRYZijLZR-Q"
	//"CxKWTzr-k6s"

	video, err := client.GetVideo("djV11Xbc914")
	if err != nil {
		log.Fatal(err)
	}

	formats := video.Formats.WithAudioChannels()

	url, err := client.GetStreamURL(video, &formats[0])
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(url)
}
```

**Comment by cybre** [comment]
The interesting I've noticed is that it only happens with music. Other types of videos don't have this problem.

**Comment by wellmoon** [comment]
same problem

**Comment by cybre** [comment]
So I've found the problem. The actions object name in `player.js` is now `$z`.  Because of this, the [following regex](https://github.com/kkdai/youtube/blob/523784c43bad823db04e1937fdad56bc97c97232/decipher.go#L199) will fail. This is because we interpolate the obj name (in this case `$z`) into the regex, but `$` is a special character. Simply escaping it fixes the issue. I have fixed this locally by escaping the parameters with `regexp.QuoteMeta`:
```
regex, err := regexp.Compile(fmt.Sprintf("(?:a=)?%s\\.(%s|%s|%s)\\(a,(\\d+)\\)", regexp.QuoteMeta(string(obj)), regexp.QuoteMeta(reverseKey), regexp.QuoteMeta(spliceKey), regexp.QuoteMeta(swapKey)))
```
@corny Is this an acceptable fix? If so, I would gladly open a PR.
